muayKenny/nomic-demo-search-app
# Kenny Refactor

# Search Page Improvement

## What I added
- Search state persists through URL parameters, enabling sharing and bookmarking
- Added configurable number of results (k) with validation
- Enhanced state management to handle input/query separately
- Added proper TypeScript interfaces for data structures
- Improved error handling through Next.js error boundaries
- Added environment variable support for API URL


## How to Test It Out
1. Try a Search
   - Enter a text query and hit search
   - You should see nearest neighbor results with their similarity scores
   - Watch for the loading state while it searches

2. Adjust Neighbor Count
   - Try changing the k value to get more/fewer similar results
   - Results should update automatically

3. Check URL Features
   - After searching, copy the URL
   - Opening in a new tab should find the same neighbors
   - Bookmarking should preserve your search
